### PR TITLE
fix: added list_files method back, matching docs

### PR DESCRIPTION
**Pull request by sg-s**
## changes

- [x] closed https://github.com/deeporiginbio/deeporigin-client/issues/80
- [x] updated docs to point users to `.api`, not `_api`
- [x] Use platform API to show human names in dataframe header 
- [x] write tests for dataframe.to_deeporigin (to be run against a live instance)
- [x] fix a bug where writing a column with missing values is not possible
- [x] if nothing is edited, use the created by field in the dataframe header
- [x] update documentation about new functions to delete workspaces
- [x] update documentation about new functions to delete databases
- [x] refactored tests to use a common config block for pytest
- [x] updates to CLI to support deleting databases using new endpoints
- [x] updates to CLI to support deleting workspaces using new endpoints
- [x] updates to CLI to support deleting rows using new endpoints 

## screenshots

Screenshot 2024-09-23 at 8 58 57 PM

### CODE REVIEW

File: docs/ref/data-hub/low-level-api.md [hunk 1]
```diff
@@ -5,16 +5,16 @@
 We use automatically generated code to implement a low-level Python SDK that connects to the Deep Origin data hub API. This page describes that API. In most cases, you will not need to know how this works, nor will you need to use it.
 
 !!! tip "Using the methods in the DeeporiginData class"
-    The `deeporigin._api` module provides wrappers that allow you to use methods in the `deeporigin_data.DeeporiginData` class, without needing to instantiate an object and configure it. Each method of this class is exposed as a function with the same name. These functions enable you to easily use the methods of the `deeporigin_data.DeeporiginData` class.
+    The `deeporigin.api` module provides wrappers that allow you to use methods in the `deeporigin_data.DeeporiginData` class, without needing to instantiate an object and configure it. Each method of this class is exposed as a function with the same name. These functions enable you to easily use the methods of the `deeporigin_data.DeeporiginData` class.
```
Comment (sg-s) [inline docs/ref/data-hub/low-level-api.md:8]: users shouldn't be using _api

File: src/data_hub/api.py [hunk 2]
```diff
@@ -151,6 +151,37 @@ def create_database(
...
+
+    if assigned_row_ids is not None:
+        filters.append(dict(assigned_row_ids=assigned_row_ids))
+
+    if is_unassigned is not None:
+        filters.append(dict(is_unassigned=is_unassigned))
+
+    return _api.list_files(filters=filters, client=client)
+
+
```
Comment (sg-s) [inline src/data_hub/api.py:184]: we're overriding list_files because the syntax for filters is cumbersome

File: tests/test_managed_data.py [hunk 3]
```diff
@@ -173,7 +173,7 @@ def test_list_files_unassigned(config):
 
 def test_list_files_assigned(config):
     files = api.list_files(
-        filters=[dict(is_unassigned=False)],
+        is_unassigned=False,
         client=config["client"],
     )
```
Comment (sg-s) [inline tests/test_managed_data.py:178]: refactored tests to work with new function signature

File: tests/test_cli_data.py [hunk 4]
```diff
@@ -26,45 +27,7 @@
 )
 
 
-@pytest.fixture(scope="session", autouse=True)
```
Comment (sg-s) [inline tests/test_cli_data.py:29]: we move this to tests.utils, so we can re-use this config block across all test modules

File: tests/test_dataframe.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+from tests.utils import clean_up_test_objects
+
+TEST_PREFIX = "tc-x1zrncgf6F-"
+TEST_DB_NAME = TEST_PREFIX + "db"
+COLUMNS = ["float", "integer"]
+
+
+@pytest.fixture(scope="session", autouse=True)
+def config(pytestconfig):
```
Comment (sg-s) [inline tests/test_dataframe.py:16]: this fixture creates a database that we can pull/operate on/write to using the DataFrame class

File: src/data_hub/api.py [hunk 6]
```diff
@@ -904,6 +904,9 @@ def get_dataframe(
 
     # remove notebook columns because they are not
     # shown in the UI as columns
+    if columns is None:
+        return None
+
```
Comment (sg-s) [inline src/data_hub/api.py:909]: if a database has no columns, then it has no data. there's nothing we can do except return None

**Review by kennyjwilli — APPROVED** [review]

File: src/platform/api.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    org_id = value["organization_id"]
+
+    headers = {
+        "accept": "application/json",
+        "authorization": f"Bearer {access_token}",
+        "cache-control": "no-cache",
+    }
+
+    response = requests.get(
+        f"https://os.deeporigin.io/api/organizations/{org_id}/users/{user_drn}",
```
[the remark this answers is not in the document]
Comment (sg-s) [reply src/platform/api.py:24]: not yet

[the remark this answers is not in the document]
Comment (sg-s) [reply src/platform/api.py:24]: but that's a good point we need to take into account environments here

[the remark this answers is not in the document]
Comment (sg-s) [reply src/platform/api.py]: fixed

File: src/platform/api.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+def get_last_edited_user_name(row):
+    """given a row object, return a human readable user name
+    of the user who last edited the row"""
+
+    if row.edited_by_user_drn is None:
+        # fall back to created by user
+        user_drn = row.created_by_user_drn.split(":")[-1]
+    else:
+        user_drn = row.edited_by_user_drn.split(":")[-1]
+    return get_user_name(user_drn)
```
[the remark this answers is not in the document]
Comment (sg-s) [reply src/platform/api.py]: fixed

**Pull request merged**